### DISCUSSION TITLE
How can I obtain a list of results found in semantic memory?

### DISCUSSION THREAD

**Discussion by wmeints** [microsoft/semantic-kernel#6079] [Q&A] [answered]
I'm building a RAG system that needs to provide references to results used in the answer. I can try to get that working with a prompt, but I want to be able to verify results by looking at the entries returned from the vector store. Is there any way I can do this in Semantic Kernel?

**Comment by dmytrostruk** [comment, marked as answer]
@wmeints If you wrap your vector store in Semantic Kernel Plugin, every function invocation from plugin can be intercepted with filter that will allow to see function result (which in your case will be vector store result) and you can extend or override result value if needed. Here are some examples:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/Concepts/Filtering/FunctionInvocationFiltering.cs

Let me know if this will cover your scenario, thanks!

**Reply by wmeints** [reply]
That's exactly what I need. Thanks!
